### PR TITLE
Added API to retrieve IP on a CNI network.

### PR DESCRIPTION
**Pull request by asridharan**
This change is [Reviewable](https://reviewable.io/reviews/mesosphere/dcos-commons/1082)

### CODE REVIEW

**Comment by asridharan** [comment]
@ArtRand can you take a look at this?

File: sdk/bootstrap/main.go [hunk 1]
```diff
@@ -5,15 +5,16 @@ import (
 	"flag"
 	"fmt"
 	// TODO switch to upstream once https://github.com/hoisie/mustache/pull/57 is merged:
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: should move the comment as well

File: sdk/bootstrap/main.go [hunk 2]
```diff
@@ -330,48 +331,43 @@ func isFile(path string) (bool, error) {
...
+	return ip.String()
 }
+
 // main
 
 func main() {
 	args := parseArgs()
 
-    libprocess_ip, found := os.LookupEnv("LIBPROCESS_IP")
+	libprocess_ip, found := os.LookupEnv("LIBPROCESS_IP")
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: there are now two fetches of LIBPROCESS_IP, one here and one in `ip.go`. should this one go away?

File: sdk/bootstrap/src/pkg/mesos/ip.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// This helper function gives the current container's IP address, when run in a
+// given network, mnt and UTS namespace.  The function first checks the presence of
+// `LIBPROCESS_IP` if present it will return this as the container's IP address
+// else it will look at the `hostname` setup for the container to pick out the
+// IP address associated with the hostname.
+func ContainerIP() (containerIP net.IP, err error) {
+	ip := os.Getenv("LIBPROCESS_IP")
+	if ip != "" {
+		containerIP = net.ParseIP(ip)
+		if containerIP == nil {
```
Comment (nickbp) [inline sdk/bootstrap/src/pkg/mesos/ip.go]: what happens when LIBPROCESS_IP is `0.0.0.0`?

Comment (asridharan) [inline sdk/bootstrap/src/pkg/mesos/ip.go]: good catch !! Should have added that case.

File: sdk/bootstrap/main.go (hunk 2 above)
Comment (asridharan) [inline sdk/bootstrap/main.go]: Not sure @ArtRand ??

Comment (asridharan) [inline sdk/bootstrap/main.go]: Actually didn't realize the logic over here. Was thinking @ArtRand might need to fix it in his patch. Will fix it in this patch itself.

**Comment by asridharan** [comment]
@ArtRand @nickbp just fixed @nickbp comments. Most of the changes are due to the formatting that my VIM plugin introduced, based on the standard go formatting template. So please don't be put of by the number of lines of change.

File: sdk/bootstrap/main.go [hunk 4]
```diff
@@ -4,16 +4,16 @@ import (
 	"bytes"
 	"flag"
 	"fmt"
-	// TODO switch to upstream once https://github.com/hoisie/mustache/pull/57 is merged:
-	"github.com/nickbp/mustache"
 	"github.com/aryann/difflib"
+	"github.com/nickbp/mustache"
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: the TODO should probably stick around...

File: sdk/bootstrap/main.go [hunk 5]
```diff
@@ -329,49 +329,37 @@ func isFile(path string) (bool, error) {
...
+	if err != nil {
+		log.Fatalf("Cannot find the container's IP address: ", err)
+	}
 
-        err := os.Setenv("LIBPROCESS_IP", libprocess_ip)
-        if err != nil {
-            log.Fatalf("Failed to SET new LIBPROCESS_IP")
-        }
-    }
+	err = os.Setenv("LIBPROCESS_IP", libprocess_ip)
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: Is this only going to apply to child processes of `bootstrap`? In practice bootstrap is run like this:

```
./bootstrap --foo && echo "other stuff here..."
```

File: sdk/bootstrap/main.go [hunk 6]
```diff
@@ -329,49 +329,37 @@ func isFile(path string) (bool, error) {
...
 
 func main() {
 	args := parseArgs()
 
-    libprocess_ip, found := os.LookupEnv("LIBPROCESS_IP")
-
-    if !found {
-        log.Fatalf("Cannot find LIBPROCESS_IP")
-    }
+	libprocess_ip, err := GetLocalIP()
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: Should this block only run when getTaskIp is enabled?

File: sdk/bootstrap/main.go [hunk 7]
```diff
@@ -389,9 +377,9 @@ func main() {
 		log.Printf("Template handling disabled via -template=false: Skipping any config templates")
 	}
 
-	if (args.installCerts) {
+	if args.installCerts {
 		installDCOSCertIntoJRE()
 	}
-    log.Printf("Local IP --> %s", GetLocalIP())
+	log.Printf("Local IP --> %s", libprocess_ip)
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: I think this could be omitted in favor of the handling within getTaskIp

**Comment by nickbp** [comment]
I'm assuming the goal would be to allow upstream to do something like:

```
export MY_IP=$(bootstrap --get-ip)
```

**Comment by nickbp** [comment]
FYI build is currently failing with:

```
Creating symlink from GOPATH=/root/go/src/github.com/mesosphere/dcos-commons to REPOPATH=/jenkins/workspace/dcos-commons/inf-pr
package pkg/mesos: unrecognized import path "pkg/mesos" (import path does not begin with hostname)
```

**Pull request merged**